CHANGELOG.md

```diff
@@ -1,3 +1,393 @@
+  kubectl delete pvc data-alertmanager-prometheus-operator-alertmanager-0 -n monitoring
+  ```
+
+3. Remove current `spec.claimRef` values to change the PV's status from Released to Available.
+
+  ```bash
+  kubectl patch pv --type json -p='[{"op": "remove", "path": "/spec/claimRef"}]' $(kubectl get pv -o jsonpath='{.items[?(@.spec.claimRef.name=="data-prometheus-prometheus-operator-prometheus-0")].metadata.name}')
+
+  kubectl patch pv --type json -p='[{"op": "remove", "path": "/spec/claimRef"}]' $(kubectl get pv -o jsonpath='{.items[?(@.spec.claimRef.name=="data-alertmanager-prometheus-operator-alertmanager-0")].metadata.name}')
+  ```
+
+  > **NOTE:** To execute the above command, the user must have a cluster wide permission.
+
+4. Make sure that the `prometheus-operator`'s `storage_class` and `prometheus.storage_size` are unchanged during the upgrade process.
+
+5. Proceed to a fresh `prometheus-operator` component installation. The new release should now re-attach your previously released PV with its content.
+
+  ```
+  lokoctl component apply prometheus-operator
+  ```
+
+  > **NOTE:** Etcd dashboard will only start showing data after the cluster is updated.
+
+6. Delete the old kubelet service.
+
+  ```bash
+  kubectl -n kube-system delete svc prometheus-operator-kubelet
+  ```
+
+7. If monitoring was enabled for `rook`, `contour`, `metallb` components, make sure you update them as well after the cluster is updated.
+
+#### Cluster update steps
+
+> **NOTE:** Updating multiple Lokomotive versions at a time is not supported. If your cluster is running a version older than `v0.5.0`, update to `v0.5.0` first and only then proceed with the update to `v0.6.0`.
+
+Please perform the following manual steps in your cluster configuration directory.
+
+1. Download the release bundle.
+
+  ```bash
+  curl -LO https://github.com/kinvolk/lokomotive/archive/v0.6.0.tar.gz
+  tar -xvzf v0.6.0.tar.gz
+  ```
+
+2. Install the Packet CCM.
+
+  If you are running Lokomotive on Equinix Metal (formerly Packet), then install Packet CCM. Export your Packet cluster's project ID and API Key.
+
+  ```bash
+  export PACKET_AUTH_TOKEN=""
+  export PACKET_PROJECT_ID=""
+
+  echo "apiKey: $PACKET_AUTH_TOKEN
+projectID: $PACKET_PROJECT_ID" > /tmp/ccm-values.yaml
+
+  helm install packet-ccm --namespace kube-system --values=/tmp/ccm-values.yaml ./lokomotive-0.6.0/assets/charts/control-plane/packet-ccm/
+  ```
+
+3. Update node config.
+
+  This script shipped with release tarball will add permanent MetalLB labels and kubelet config to use CCM.
+
+  > **NOTE:** Please edit this script to disable updating certain nodes. Modify `update_other_nodes` function as required.
+
+  ```bash
+  UPDATE_BOOTSTRAP_COMPONENTS=false
+  ./lokomotive-0.6.0/scripts/update/0.5-0.6/update.sh $UPDATE_BOOTSTRAP_COMPONENTS
+  ```
+
+4. If you're using the self-hosted kubelet, apply the `--cloud-provider` flag to it.
```

Comment:
There is no need to do that manually I think. `lokoctl cluster apply` should apply that automatically.

Reply:
This step is being run before the cluster is updated, hence I included this step.

Reply:
But why do we need to run this before cluster update?

Reply:
Because the self-hosted kubelets and bootstrap kubelets are on the same config.

Reply:
So long you don't remove the Node object, it shouldn't matter. I don't think there is a strict requirement to do that right after patching bootstrap kubelet. It simplifies and speeds up the upgrade process, which is already very long and complicated...

Reply:
If a user missed to use the `--upgrade-kubelets` with `lokoctl cluster apply` it is highly possible that the subsequent nodes added won't join the cluster, so this steps acutally guards against that mistake.

Reply:
Then we should require using `--upgrade-kubelets`?

Reply:
Yes, there is a note saying if you are using self-hosted kubelet then use this flag.

Reply:
Then I think it should be fine to remove this step if we _require_ using `--upgrade-kubelets`.

Reply:
Given https://github.com/kinvolk/lokomotive/issues/1313 I think we should not require `--upgrade-kubelets` until that's solved.

But I wonder if we can include this step in the update script checking if the self-hosted kubelet is running.

Reply:
Yeah with #1313 we will need this manual step.